spacy/tokens/span_group.pyx

```diff
@@ -128,6 +136,65 @@ cdef class SpanGroup:
             i += size
         return Span.cinit(self.doc, self.c[i])
 
+    def __delitem__(self, int i) :
+        """Delete a span from the group at index i
+
+        i (int): The item index.
+
+        DOCS: https://spacy.io/api/spangroup#delitem
+        """
+        cdef int size = self.c.size()
+        if i < -size or i >= size:
+            raise IndexError(f"list index {i} out of range")
+        if i < 0:
+            i += size
```

Comment:
It looks like we're copying the check & adjustement of index `i` across various functions - perhaps we could use a simple private helper for this?

Reply:
Done.